src/django_nh3/utils.py

```diff
@@ -16,35 +18,177 @@ def get_nh3_default_options() -> dict[str, Any]:
+        # Returns None to remove the attribute, or a value to use
+        "NH3_ALLOWED_ATTRIBUTES_FILTER": "attribute_filter",
+        # Configures the handling of HTML comments, defaults to True
         "NH3_STRIP_COMMENTS": "strip_comments",
+        # Configures a rel attribute that will be added on links, defaults to
+        # noopener noreferrer. To turn on rel-insertion, pass a space-separated
+        # list. If rel is in the generic or tag attributes, this must be set to
+        # None
+        # Common rel values to include:
+        # noopener
+        # noreferrer
+        # nofollow
+        "NH3_LINK_REL": "link_rel",
+        # Sets the prefix of attributes that are allowed on any tag
+        "NH3_ALLOWED_GENERIC_ATTRIBUTE_PREFIXES": "generic_attribute_prefixes",
+        # Sets the values of HTML attributes that are allowed on specific tags.
+        # The value is structured as a map from tag names to a map from
+        # attribute names to a set of attribute values. If a tag is not itself
+        # whitelisted, adding entries to this map will do nothing.
+        "NH3_ALLOWED_TAG_ATTRIBUTE_VALUES": "tag_attribute_values",
+        # Sets the values of HTML attributes that are to be set on specific
+        # tags. The value is structured as a map from tag names to a map from
+        # attribute names to an attribute value. If a tag is not itself
+        # whitelisted, adding entries to this map will do nothing.
+        "NH3_SET_TAG_ATTRIBUTE_VALUES": "set_tag_attribute_values",
+        # Sets the URL schemes permitted on href and src attributes
+        "NH3_ALLOWED_URL_SCHEMES": "url_schemes",
     }
 
-    for setting, kwarg in nh3_settings.items():
-        if hasattr(settings, setting):
-            attr = getattr(settings, setting)
+    return {
+        kwarg: getattr(settings, setting_name)
+        for setting_name, kwarg in nh3_settings.items()
+        if hasattr(settings, setting_name)
+    }
+
+
+def normalize_nh3_options(  # noqa: C901, PLR0912
+    options: dict[str, Any],
+) -> dict[str, Any]:
+    nh3_args: dict[str, Any] = {}
+    for kwarg_name, kwarg_value in options.items():
+        value = kwarg_value
+
+        # Convert from general iterables to sets
+        if kwarg_name in [
+            "tags",
+            "clean_content_tags",
+            "generic_attribute_prefixes",
+            "url_schemes",
+        ]:
+            value = set(value)
+
+        elif kwarg_name == "attributes":
+            copy_dict = value.copy()
+            for tag, attributes in value.items():
+                copy_dict[tag] = set(attributes)
+            value = copy_dict
+
+        elif kwarg_name == "attribute_filter":
+            if callable(value):
+                pass
+            elif isinstance(value, str):
+                value = import_string(value)
+
+        elif kwarg_name == "strip_comments":
+            value = bool(value)
```

Comment:
Is this value defined in python and therefore something we know will be `True` or `False`, rather than something that might become a string?

Reply:
The intention behind this function is that we don't have to care - after settings pass through this function, they will be what we need/expect them to be.

But to answer your question directly: it will come from one of two places. Either it will come from the configured defaults (eg: the Django project `settings.py`), or it will come from values set on a model or form field. In both places, it should be a Python value, but it is difficult to predict how users will use and misuse libraries sometimes, so I included it. I figured it we were checking everything else, we might as well check this value as well.